rust/lance/src/dataset/scanner.rs

```diff
@@ -1969,7 +1969,7 @@ mod test {
         let test_dir = tempdir().unwrap();
         let test_uri = test_dir.path().to_str().unwrap();
         let mut data_gen = BatchGenerator::new().col(Box::new(
-            IncrementingInt32::new().named("filter_me".to_owned()),
+            IncrementingInt32::new().named("Filter_me".to_owned()),
```

Comment:
Could we also add a column nested in a struct, and verify we can reference it in the filter by wrapping the segments in backtick?

Something like:

```sql
`nested with space`.`inner with space` < 2
```

(This example is from our existing SQL docs)

Reply:
Added a test, see https://github.com/lancedb/lance/pull/1232/commits/9959ceef16aea99fb3df184eecfedf15bd7f900f